rioxarray/rioxarray.py

```diff
@@ -1078,3 +1079,90 @@ def transform_bounds(self, dst_crs, densify_pts=21, recalc=False):
         return rasterio.warp.transform_bounds(
             self.crs, dst_crs, *self.bounds(recalc=recalc), densify_pts=densify_pts
         )
+
+    def write_gcps(self, gcps, grid_mapping_name=None, inplace=False):
+        """
+        Write the GroundControlPoints to the dataset.
+
+        https://rasterio.readthedocs.io/en/latest/topics/georeferencing.html#ground-control-points
+
+        Parameters
+        ----------
+        gcp: (list of rasterio.GroundControlPoints, corresponding crs)
+            The Ground Control Points to integrate to the dataset.
+        grid_mapping_name: str, optional
+            Name of the grid_mapping coordinate to store the GCPs information in.
+            Default is the grid_mapping name of the dataset.
+        inplace: bool, optional
+            If True, it will write to the existing dataset. Default is False.
+
+        Returns
+        -------
+        :obj:`xarray.Dataset` | :obj:`xarray.DataArray`:
+            Modified dataset with Ground Control Points written.
+        """
+        grid_mapping_name = (
+            self.grid_mapping if grid_mapping_name is None else grid_mapping_name
+        )
+        data_obj = self._get_obj(inplace=True)
+
+        gcps, gcp_crs = gcps
+        data_obj = data_obj.rio.write_crs(
+            gcp_crs, grid_mapping_name=grid_mapping_name, inplace=inplace
+        )
+        geojson_gcps = _convert_gcps_to_geojson(gcps)
+        data_obj.coords[grid_mapping_name].attrs["gcps"] = geojson_gcps
+        return data_obj
+
+    def get_gcps(self):
+        """
+        Get the GroundControlPoints from the dataset.
+
+        https://rasterio.readthedocs.io/en/latest/topics/georeferencing.html#ground-control-points
+
+        Returns
+        ----------
+        (list of rasterio.GroundControlPoints, corresponding crs)
```

Comment:
I see you have the corresponding crs listed here in the docstring, but not on the output. Is that the behavior you intended to have? If so, it would match what rasterio does: https://rasterio.readthedocs.io/en/latest/api/rasterio._base.html#rasterio._base.DatasetBase.get_gcps